app/views/admin/products/_details_fields.haml

```diff
@@ -0,0 +1,24 @@
+= f.input :name, hint: text(".hints.name", field: f.object.class.name.downcase)
+
+- url_name_hint = begin send("facility_#{f.object.class.name.downcase}_url", current_facility, f.object) rescue send("facility_#{f.object.class.name.downcase}_url", current_facility, "url-name") end
```

Comment:
Agreed on not taking on all view cleanup right now, but I'd make an exception for this because it's got a `rescue` in it.